support/metrics/metricsTracker.go

```diff
@@ -0,0 +1,177 @@
+type event struct {
+	UserID    string      `json:"user_id"`
+	DeviceID  string      `json:"device_id"`
+	EventType string      `json:"event_type"`
+	Props     interface{} `json:"event_properties"`
+}
+
+// props holds the properties that we need for all Amplitude events.
+// This lives on the `MetricsTracker` struct.
+// TODO DS Add geodata.
+// TODO DS Add cloud server information.
+type commonProps struct {
+	CliVersion                string    `json:"cli_version"`
+	Goos                      string    `json:"goos"`
+	Goarch                    string    `json:"goarch"`
+	Goarm                     string    `json:"goarm"`
+	GuiVersion                string    `json:"gui_version"`
+	Strategy                  string    `json:"strategy"`
+	UpdateTimeIntervalSeconds int32     `json:"update_time_interval_seconds"`
+	Exchange                  string    `json:"exchange"`
+	TradingPair               string    `json:"trading_pair"`
+	SessionID                 uuid.UUID `json:"session_id"`
+	SecondsSinceStart         float64   `json:"seconds_since_start"`
+}
+
+// updateProps holds the properties for the update Amplitude event.
+type updateProps struct {
+	commonProps
+	Success bool `json:"success"`
+}
+
+// deleteProps holds the properties for the delete Amplitude event.
+// TODO DS StackTrace may need to be a message instead of or in addition to a
+// stack trace. The goal is to get crash logs, Amplitude may not enable this.
+type deleteProps struct {
+	commonProps
+	Exit       bool   `json:"exit"`
+	StackTrace string `json:"stack_trace"`
+}
+
+// MakeMetricsTracker is a factory method to create a `metrics.Tracker`.
+func MakeMetricsTracker(
+	userID string,
+	apiKey string,
+	client *http.Client,
+	start time.Time,
+	version string,
+	goos string,
+	goarch string,
+	goarm string,
+	guiVersion string,
+	strategy string,
+	updateTimeIntervalSeconds int32,
+	exchange string,
+	tradingPair string,
+) (*MetricsTracker, error) {
+	sessionID, e := uuid.NewUUID()
+	if e != nil {
+		return nil, fmt.Errorf("could not generate uuid with error %s", e)
+	}
+	props := commonProps{
+		CliVersion:                version,
+		Goos:                      goos,
+		Goarch:                    goarch,
+		Goarm:                     goarm,
+		GuiVersion:                guiVersion,
+		Strategy:                  strategy,
+		UpdateTimeIntervalSeconds: updateTimeIntervalSeconds,
+		Exchange:                  exchange,
+		TradingPair:               tradingPair,
+		SessionID:                 sessionID,
+	}
+
+	return &MetricsTracker{
+		userID: userID,
+		client: client,
+		apiKey: apiKey,
+		props:  props,
+		start:  start,
+	}, nil
+}
+
+// SendStartupEvent sends the startup Amplitude event.
+func (mt *MetricsTracker) SendStartupEvent() error {
+	return mt.sendEvent(startupEventName, mt.props)
+}
+
+// SendUpdateEvent sends the update Amplitude event.
+func (mt *MetricsTracker) SendUpdateEvent(now time.Time, success bool) error {
```

Comment:
TODO NS: also take in `millisForUpdate` property and add to struct accordingly

Reply:
moving to #516

Reply:
Added a TODO for this on the `commonProps` struct, seems like it requires a bit more implementation work and has been added to a follow-up issue as is.